### PR TITLE
DigitalOcean OAuth UI

### PR DESCRIPTION
**Pull request by alalamav**
* Implements the UI for the new DigitalOcean OAuth flow:
screen shot 2018-07-11 at 1 37 35 pm
screen shot 2018-07-11 at 1 42 10 pm
screen shot 2018-07-11 at 1 42 40 pm
screen shot 2018-07-11 at 1 43 05 pm
screen shot 2018-07-11 at 1 44 44 pm
screen shot 2018-07-11 at 1 36 44 pm
screen shot 2018-07-11 at 3 01 47 pm

* UI refactor:
  * Creates a reusable component, `outline-step-view` to display the steps UI (intro, OAuth, server creation).
  * Deprecates `outline-server-creator`, moving its functionality to app root.
  * Refactors the app drawer to use paper-listbox, which allows us to remove some JavaScript code.
* Supersedes #191.
* FYI @ohnocassio, @sandrigo.

### CODE REVIEW

File: src/server_manager/web_app/app.ts [hunk 1]
```diff
@@ -373,18 +372,17 @@ export class App {
...
-            ['Cancel'])
-        .then(() => {
-          session.cancel();
-        });
+    const handleOauthFlowCanceled = () => {
+      session.cancel();
+      this.clearCredentialsAndShowIntro();
+      this.appRoot.removeEventListener('DigitalOceanOauthCancelRequested', handleOauthFlowCanceled);
+    };
+    this.appRoot.addEventListener('DigitalOceanOauthCancelRequested', handleOauthFlowCanceled);
```
Comment (fortuna) [inline src/server_manager/web_app/app.ts]: Same here. The listener may never be removed.

File: src/server_manager/web_app/app.ts [hunk 2]
```diff
@@ -240,53 +241,51 @@ export class App {
...
     });
 
-    authEvents.once('cancel', () => {
+    const handleOauthFlowCanceled = () => {
       cancelled = true;
       this.clearCredentialsAndShowIntro();
-    });
+      this.appRoot.removeEventListener('DigitalOceanOauthCancelRequested', handleOauthFlowCanceled);
+    };
+    this.appRoot.addEventListener('DigitalOceanOauthCancelRequested', handleOauthFlowCanceled);
```
[the remark this answers is not in the document]
Comment (fortuna) [reply src/server_manager/web_app/app.ts]: Here is a possibility: pass a `onCancel` callback to `this.appRoot.getDigitalOceanOauthFlow()`.

That would get reset each time you call the method.

File: src/server_manager/web_app/app.ts [hunk 3]
```diff
@@ -373,18 +372,17 @@ export class App {
   }
 
   private connectToDigitalOcean() {
+    const oauthUi = this.appRoot.getAndShowDigitalOceanOauthFlow();
```
Comment (fortuna) [inline src/server_manager/web_app/app.ts]: We can pass a `onCancel` callback here too. You'll need to move this to after `runDigitalOceanOauth`

File: src/server_manager/ui_components/outline-intro-step.html [hunk 4]
```diff
@@ -97,68 +103,79 @@
...
-        color: rgba(255, 255, 255, 0.54);
+        color: rgba(0,0,0,0.87);
+        letter-spacing: 0;
+        line-height: 32px;
         margin: 24px 0px 0px 50px;
       }
       .card .cardRightColumn ul {
         margin-top: 0px;
-        color: rgba(255, 255, 255, 0.87);
+        /*color: rgba(255, 255, 255, 0.87);*/
```
[the remark this answers is not in the document]
Comment (alalamav) [reply src/server_manager/ui_components/outline-intro-step.html]: Done.

File: src/server_manager/ui_components/outline-step-view.html [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        <div class="step-title"><slot name="step-title"></slot></div>
+        <div class="step-description"><slot name="step-description"></slot></div>
+      </div>
+    </div>
+    <slot></slot>
+  </template>
+  <script>
+    Polymer({
+      is: 'outline-step-view',
+      properties: {
```
[the remark this answers is not in the document]
Comment (alalamav) [reply src/server_manager/ui_components/outline-step-view.html]: Done.

File: src/server_manager/web_app/app.ts [hunk 6]
```diff
@@ -794,4 +792,8 @@ export class App {
       this.showCreateServer();
     });
   }
+
+  private sleep(ms: number): Promise<void> {
```
[the remark this answers is not in the document]
Comment (alalamav) [reply src/server_manager/web_app/app.ts]: Done.

File: src/server_manager/web_app/app.ts [hunk 7]
```diff
@@ -240,53 +241,51 @@ export class App {
...
+        let maybeSleep = Promise.resolve();
+        if (activatingAccount) {
+          // Show the 'account active' screen for a few seconds if the account was activated during
+          // this session.
+          oauthUi.showAccountActive();
+          maybeSleep = this.sleep(1500);
+        }
+        maybeSleep.then(() => {
+          this.digitalOceanRepository = this.createDigitalOceanServerRepository(doSession);
+          this.digitalOceanRepository.listServers()
```
[the remark this answers is not in the document]
Comment (alalamav) [reply src/server_manager/web_app/app.ts]: Done.

File: src/server_manager/web_app/app.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (alalamav) [reply src/server_manager/web_app/app.ts]: Done.

File: src/server_manager/web_app/app.ts (hunk 1 above)
Comment (alalamav) [reply src/server_manager/web_app/app.ts]: Done.

**Comment by alalamav** [comment]
Thank you for the review!

File: src/server_manager/web_app/app.ts [hunk 8]
```diff
@@ -209,10 +213,14 @@ export class App {
...
     const doSession = this.createDigitalOceanSession(accessToken);
-    const oauthUi = this.appRoot.getDigitalOceanOauthFlow();
     const authEvents = new events.EventEmitter();
     let cancelled = false;
     let activatingAccount = false;
+    const handleOauthFlowCanceled = () => {
+      cancelled = true;
+      this.clearCredentialsAndShowIntro();
+    };
+    const oauthUi = this.appRoot.getDigitalOceanOauthFlow(handleOauthFlowCanceled);
```
Comment (fortuna) [inline src/server_manager/web_app/app.ts]: We are not running the Oauth flow here. It happened already. We are verifying the state of the account. The current name gives the impression we are running the Oauth flow. Let's rename it to better express what we are doing. For example `cancelAccountStateVerification` or `cancelWaitForActiveAccount`

File: src/server_manager/web_app/app.ts [hunk 9]
```diff
@@ -372,14 +375,12 @@ export class App {
...
   private connectToDigitalOcean() {
-    const oauthUi = this.appRoot.getAndShowDigitalOceanOauthFlow();
     const session = runDigitalOceanOauth();
     const handleOauthFlowCanceled = () => {
       session.cancel();
       this.clearCredentialsAndShowIntro();
-      this.appRoot.removeEventListener('DigitalOceanOauthCancelRequested', handleOauthFlowCanceled);
     };
-    this.appRoot.addEventListener('DigitalOceanOauthCancelRequested', handleOauthFlowCanceled);
+    const oauthUi = this.appRoot.getAndShowDigitalOceanOauthFlow(handleOauthFlowCanceled);
```
Comment (fortuna) [inline src/server_manager/web_app/app.ts]: I think `oauthUi` is never used

**Pull request merged**